plugins/poll/assets/javascripts/widgets/discourse-poll.js.es6

```diff
@@ -636,13 +639,26 @@ function pieChartConfig(data, labels, opts = {}) {
           backgroundColor: getColors(data.length)
         }
       ],
-      labels
+      labels: strippedLabels
     },
     options: {
       responsive: true,
       aspectRatio,
       animation: { duration: 400 },
-      legend: { display: displayLegend }
+      legend: { display: false },
+      legendCallback: function(chart) {
+        let text = [];
+        for (let i = 0; i < labels.length; i++) {
+          text.push(
```

Comment:
As much as possible I think it's better create DOM objects here instead of string string concatenation.

https://developer.mozilla.org/en-US/docs/Web/API/Document/createElement

Reply:
I believe this introducing more complexity because we are inside a widget.

I changed the legendCallback to this: 
```javascript
      legendCallback: function(chart) {
        let legendsContainer = document.createElement('div')
        for (let i = 0; i < labels.length; i++) {
          let legend = document.createElement("div")
          let swatch = document.createElement("span")
          legend.classList.add('legend')
          swatch.classList.add('swatch')
          swatch.style.backgroundColor = chart.data.datasets[0].backgroundColor[i]
          let legendLabel = document.createElement('span')
          legendLabel.innerHTML = labels[i]
          legend.appendChild(swatch)
          legend.appendChild(option)
          legendsContainer.appendChild(legend)
        }
        return legendsContainer
      }
```

And then inside the widget
```javascript
      document.getElementById(
        `poll-results-legend-${attrs.id}`
      ).appendChild(chart.generateLegend());
```

But the problem is now that those DOM  elements are being held outside of scope of the widget, so they are not being garbage collected. Whenever the widget is re-rendered, the legend is duplicated.

![Screenshot from 2020-02-10 15-57-42](https://user-images.githubusercontent.com/16214023/74194237-8312d200-4c1e-11ea-8c56-4248685e4d75.png)

I think returning an HTML string here is better than doing manual cleanup. What do you think @jjaffeux ?

Reply:
there's probably a way but chart.js doesnt make it easy for sure... Let's give up on this. Thanks for trying.